scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.jsx

```diff
@@ -106,9 +105,7 @@ SpellcheckerContextMenuComponent.propTypes = {
  * @description Maps the values from state to the component callback type props.
  */
 const mapDispatchToProps = (dispatch) => ({
-    addWord: (word) => dispatch(actions.ignoreWord(word)),
-    ignoreWord: (word) => dispatch(actions.addWord(word)),
-    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0),
+    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0)
```

Comment:
Actually, you would have to change [this function](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/spellcheck/spellcheck.js#L414) too, so that it returns a promise in both outcomes...

Ping me on Slack and I'm happy to help with this...